src/Core/ServiceWrapper/Program.cs

```diff
@@ -23,18 +23,38 @@
 using winsw.Util;
 using WMI;
 using ServiceType = WMI.ServiceType;
+using CommandLine;
+using System.Reflection;
+using System.Linq;
+using winsw.CLI;
 
 namespace winsw
 {
     public static class Program
     {
-        private static readonly ILog Log = LogManager.GetLogger(typeof(Program));
+        public static readonly ILog Log = LogManager.GetLogger(typeof(Program));
+
+        public static bool elevated;
+
+        public static CLICommand cliOption;
 
         public static int Main(string[] args)
         {
             try
             {
-                Run(args);
+                if (!handleIncorrectArgument(args))
+                {
+                    Console.WriteLine("Invalid command [" + args[0] + "]");
+                    PrintAvailableCommands();
+                    return -1;
+                }
+
+                var types = LoadVerbs();
+
+                Parser.Default.ParseArguments(args, types)
+                        .WithParsed(RunParsed)
```

Comment:
I just took a look at the docs. Why don't you use the generic version of `WithParsed` in the first place?